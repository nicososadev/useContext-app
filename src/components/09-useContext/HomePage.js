import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const HomePage = () => {

    const { user } = useContext(userContext)

    return (
        <div>
            <h1>Homepage</h1>
            <hr />

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>
            
        </div>
    )
}
