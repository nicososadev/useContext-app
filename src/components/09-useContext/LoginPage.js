import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const LoginPage = () => {

    const { setUser } = useContext(userContext)

    return (

        <div>
            <h1>Login</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={ () => setUser({
                    id: 1,
                    name: 'Nicolas'
                })}
            >
                Login
            </button>
        </div>
    )
}
