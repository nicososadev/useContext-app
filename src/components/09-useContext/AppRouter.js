import React from 'react'
import { BrowserRouter as Router, Switch, Route } from "react-router-dom"
import { AboutPage } from './AboutPage'
import { HomePage } from './HomePage'
import { LoginPage } from './LoginPage'
import { NavBar } from './NavBar'

export const AppRouter = () => {

    return (
        <Router>
            <div>

                <NavBar />
                <div className="container">
                    <Switch>
                        <Route exact  path="/" component={ HomePage }/>

                        <Route exact path="/login" component={ LoginPage }/>
                        <Route exact path="/about" component={ AboutPage }/>

                        <Route component={ HomePage } />
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
