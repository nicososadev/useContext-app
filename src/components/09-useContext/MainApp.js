import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { userContext } from './UserContext'

export const MainApp = () => {
    
    const [user, setUser] = useState({})

    return (
        <userContext.Provider value={{
            user,
            setUser
        }}>

            <AppRouter />

        </userContext.Provider>
    )
}
