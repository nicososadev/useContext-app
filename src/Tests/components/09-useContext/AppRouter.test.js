import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { AppRouter } from '../../../components/09-useContext/AppRouter'
import { userContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente AppRouter', () => {
    
    const user = {
        id: 0,
        name: 'Prueba'
    }

    const wrapper = mount(
        <userContext.Provider value={{ user }}>

            <AppRouter />)

        </userContext.Provider>  
    )

    test('Debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })
    
})
