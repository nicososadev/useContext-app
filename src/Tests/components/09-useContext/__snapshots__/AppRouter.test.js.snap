// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter Debe mostrarse correctamente 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/useContext-app"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/useContext-app"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </nav>
    <div
      className="container"
    >
      <div>
        <h1>
          Homepage
        </h1>
        <hr />
        <pre>
          {
   "id": 0,
   "name": "Prueba"
}
        </pre>
      </div>
    </div>
  </div>,
  ")",
]
`;
